aws-typescript/tests/infra.spec.ts

```diff
@@ -0,0 +1,51 @@
+import * as pulumi from "@pulumi/pulumi";
+import "jest";
+
+// Test helper to convert a Pulumi Output to a Promise.
+// This should only be used in tests.
+function promiseOf<T>(output: pulumi.Output<T>): Promise<T> {
+    return new Promise(resolve => output.apply(resolve));
```

Comment:
I experimented with custom jest matchers here https://github.com/pulumi/templates/pull/868

This keeps the "magic" of Outputs, and avoids the (slightly wrong) conversion to promises here.

Reply:
I am not arguing for including these custom matchers in this PR. I was curious to see what it would look like. I think it's something we can follow up with if there are is enough demand to warrant the extra work. We could package the matchers in a library, that can be (optionally) used for writing tests.